### ISSUE TITLE
Wrong error using requires on block-opening-brace-space-before

### ISSUE THREAD

**Issue by masakk1** [vala-lang/vala-lint#197]
The following function will throw a `Unexpected line break before "{" block-opening-brace-space-before`. It should probably allow for a new-line if requires or ensures are preset.

While possible to use a single space, it's a little weird

## Current behaviour
This fails `block-opening-brace-space-before`:
```vala
double method_name (int x, double d)
        requires (x > 0 && x < 10)
        requires (d >= 0.0 && d <= 1.0)
        ensures (result >= 0.0 && result <= 10.0)
{
    return d * x;
}
```

This passes:
```vala
double method_name (int x, double d)
        requires (x > 0 && x < 10)
        requires (d >= 0.0 && d <= 1.0)
        ensures (result >= 0.0 && result <= 10.0) {
    return d * x;
}
```

### Temporary solution
For now, we could just disable this check for the linter.
This is how it looks like:
```vala
private ArrayList<File> get_children (File folder, ArrayList<File> _children = new ArrayList<File> ())
        requires (folder.query_file_type (FileQueryInfoFlags.NONE, null) == FileType.DIRECTORY) // vala-lint:block-opening-brace-space-before
        {
            // .......
            return _children;
        }
```

## Soltuion
Ideally have an exception for contrat programming keywords like requires and ensures. Even the documentation on vala.dev styles the code this way:
Reference: [Contract Programming on docs.vala.dev](https://docs.vala.dev/tutorials/programming-language/main/04-00-advanced-features/04-01-assertions-and-contract-programming.html#_4-1-assertions-and-contract-programming)
